pkg/process/handler.go

```diff
@@ -2,34 +2,175 @@ package process
 
 import (
 	"github.com/AsterNighT/software-engineering-backend/api"
-	"github.com/labstack/echo/v4"
 	"net/http"
 )
 
 type RegistrationHandler struct {
 }
 
 // @Summary get all departments
-// @Tags departments
+// @Tags patient
 // @Description display all departments of a hospital
 // @Produce json
-// @Success 200 {object} api.ReturnedData{data=[]Department}
-// @Router /departments [GET]
+// @Success 200 {array} api.ReturnedData{data=[]Department}
+// @Router /patient/departments [GET]
 func (h *RegistrationHandler) GetAllDepartments(c echo.Context) error {
 
 	c.Logger().Debug("hello world")
 	return c.JSON(http.StatusCreated, api.Return("pong", nil))
 }
 
 // @Summary get a department by its ID
-// @Tags departments
+// @Tags patient
 // @Description return a department's details by its ID
 // @Param department_id path uint true "Department ID"
 // @Produce json
 // @Success 200 {object} api.ReturnedData{data=Department}
-// @Router /departments/{department_id} [GET]
+// @Router /patient/departments/{department_id} [GET]
 func (h *RegistrationHandler) GetDepartmentByID(c echo.Context) error {
 
 	c.Logger().Debug("hello world")
 	return c.JSON(http.StatusCreated, api.Return("pong", nil))
 }
+
+// @Summary patient register
+// @Tags patient
+// @Description return registration state
+// @Param patient_id, department_id
+// @Produce json
+// @Success 200 {string} api.ReturnedData{data="register success"}
+// @Router /patient/registration/create [POST]
```

Comment:
And here

Reply:
After asking the view of Ziyue Pan, We believe that the attributes contained in the returned objects from patients and doctors are different. In order to facilitate the back-end design, we can directly judge whether the view is that of the patient or the doctor based on the URL.

Reply:
So why don't we separate them to `doctor/registration` and `patient/registration`? That does not seem right to me if a patient view registration does not require no patientID but a doctor view registrations does.